src/aggregation/metric/stats.rs

```diff
@@ -143,12 +300,266 @@ impl IntermediateStats {
+/// intermediate results.
+#[derive(Clone, Debug, PartialEq, Serialize, Deserialize)]
+pub struct IntermediateExtendedStats {
+    intermediate_stats: IntermediateStats,
+    /// The number of extracted values.
+    // The sum of square values, it's referred as M2 in Welford's online algorithm
+    sum_of_squares: f64,
+    // The sum of square values as computed by elastic search
+    sum_of_squares_elastic: f64,
+    /// delta for sum of squares  as computed by elastic search needed for the Kahan algorithm
+    delta_sum_for_squares_elastic: f64,
+    // The mean an intermediate value need for calculating the variance
+    // as per [Welford's online algorithm](https://en.wikipedia.org/wiki/Algorithms_for_calculating_variance#Welford's_online_algorithm)
+    mean: f64,
+    // the value used for computing standard deviation bounds
+    sigma: f64,
+}
+
+impl Default for IntermediateExtendedStats {
+    fn default() -> Self {
+        Self {
+            intermediate_stats: IntermediateStats::default(),
+            sum_of_squares: 0.0,
+            sum_of_squares_elastic: 0.0,
+            delta_sum_for_squares_elastic: 0.0,
+            mean: 0.0,
+            sigma: 2.0,
+        }
+    }
+}
+
+impl IntermediateExtendedStats {
+    /// Creates a new IntermediateExtendedStats using an option
+    /// containing the sigma to be used for calculating bound values.
+    pub fn with_sigma(sigma: Option<f64>) -> Self {
+        Self {
+            intermediate_stats: IntermediateStats::default(),
+            sum_of_squares: 0.0,
+            sum_of_squares_elastic: 0.0,
+            delta_sum_for_squares_elastic: 0.0,
+            mean: 0.0,
+            sigma: sigma.unwrap_or(2.0),
+        }
+    }
+    /// Merges the other stats intermediate result into self.
+    pub fn merge_fruits(&mut self, other: IntermediateExtendedStats) {
+        if other.intermediate_stats.count != 0 {
+            if self.intermediate_stats.count == 0 {
+                self.sum_of_squares = other.sum_of_squares;
+                self.sum_of_squares_elastic = other.sum_of_squares_elastic;
+                self.intermediate_stats = other.intermediate_stats;
+                self.mean = other.mean;
+                self.delta_sum_for_squares_elastic = other.delta_sum_for_squares_elastic
+            } else {
+                if other.intermediate_stats.count == 1 {
+                    self.collect(other.intermediate_stats.sum);
+                } else if self.intermediate_stats.count == 1 {
+                    let sum = self.intermediate_stats.sum;
+                    self.intermediate_stats = other.intermediate_stats;
+                    self.sum_of_squares = other.sum_of_squares;
+                    self.sum_of_squares_elastic = other.sum_of_squares_elastic;
+                    self.mean = other.mean;
+                    self.delta_sum_for_squares_elastic = other.delta_sum_for_squares_elastic;
+                    self.collect(sum);
+                } else {
```

Comment:
these cases are way to special and should be handled by the generic version

Reply:
Sorry, but I don't understand it, I try to better clarify the current situation. 

I created a version with a new segment collector for extended stats which reduces the extra memory increase, but I did not pushed yet, because I'm waiting for a response about kahan summation. In the new version the generic component is gone, so I cannot apply the change.

If you were referring to some other modification I ask you for some more details.

Reply:
yes, we can use kahan summation.

What I mean is that these branches, they complexity they add, instead using the version that can handle all counts, needs to be justified. 

```rust
if other.intermediate_stats.count == 1 {
    self.collect(other.intermediate_stats.sum);
} else if self.intermediate_stats.count == 1 {
    let sum = self.intermediate_stats.sum;
    self.intermediate_stats = other.intermediate_stats;
    self.sum_of_squares = other.sum_of_squares;
    self.sum_of_squares_elastic = other.sum_of_squares_elastic;
    self.mean = other.mean;
    self.delta_sum_for_squares_elastic = other.delta_sum_for_squares_elastic;
    self.collect(sum);
```